File: Api/Models/Dispatch.cs
using Api.Models.Common;
using Api.Models.Enums;

namespace Api.Models;

public sealed class Dispatch : AuditableEntity
{
    public int Id { get; set; }
    public int OccurrenceId { get; set; }
    public Occurrence Occurrence { get; set; } = null!;
    public int UnitId { get; set; }
    public Unit Unit { get; set; } = null!;
    public int ConfirmedByUserId { get; set; }
    public User ConfirmedByUser { get; set; } = null!;
    public DispatchStatus Status { get; set; } = DispatchStatus.ATRIBUIDO;
    public DateTime? AcceptedAt { get; set; }
    public DateTime? ArrivedAt { get; set; }
    public DateTime? ServiceStartedAt { get; set; }
    public DateTime? CompletedAt { get; set; }
    public DateTime? CancelledAt { get; set; }
}

File: WebApp/src/interfaces/occurrenceInterfaces/Dispatch.ts
import type { OccurrenceCreator } from './Occurrence';

export interface DispatchRead {
  id: number;
  unitId: number;
  unitName: string;
  service: string;
  unitStatus: 'RESERVADA' | 'DESLOCAMENTO' | 'EM_ATENDIMENTO' | 'DISPONIVEL';
  status: 'ATRIBUIDO' | 'ACEITO' | 'NO_LOCAL' | 'EM_ATENDIMENTO' | 'CONCLUIDO' | 'CANCELADO';
  createdAt: string;
  updatedAt: string;
  acceptedAt?: string;
  arrivedAt?: string;
  serviceStartedAt?: string;
  completedAt?: string;
  cancelledAt?: string;
}

export interface DispatchConfirmationRead {
  occurrenceId: number;
  occurrenceStatus: 'DESPACHADA';
  confirmedAt: string;
  confirmedBy: OccurrenceCreator;
  dispatches: DispatchRead[];
}
